File: python/clase-09/final/schemas.py
from __future__ import annotations

from pydantic import BaseModel


class InvoiceItem(BaseModel):
    description: str
    quantity: float | None = None
    unit_price: float | None = None
    total: float


class InvoiceData(BaseModel):
    provider: str | None
    date: str | None
    currency: str | None
    total: float | None
    items: list[InvoiceItem]

File: typescript/clase-09/final/schemas.ts
import { z } from "zod";

export const InvoiceData = z.object({
  provider: z.string().nullable(),
  date: z.string().nullable(),
  currency: z.string().nullable(),
  total: z.number().nullable(),
  items: z.array(z.object({
    description: z.string(),
    quantity: z.number().nullable().optional(),
    unit_price: z.number().nullable().optional(),
    total: z.number(),
  })),
});
